sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs

```diff
@@ -34,6 +34,19 @@ public class BlobUriBuilder
         /// </summary>
         private readonly bool _isPathStyleUri;
 
+        /// <summary>
+        /// Whether to preserve trailing slashes on blob names when constructing a URI with this instance.
+        /// </summary>
+        public bool PreserveTrailingSlash { get; }
+
+        /// <summary>
+        /// Default value for <see cref="PreserveTrailingSlash"/> to construct an instance
+        /// of <see cref="BlobUriBuilder"/> with when unspecified. This is also the value
+        /// used by <see cref="BlobContainerClient.GetBlobClient(string)"/> and similar
+        /// methods. Defaults to false.
+        /// </summary>
+        public static bool PreserveTrailingSlashDefault { get; set; }
```

Comment:
@JoshLove-msft @tg-msft @kasobol-msft Looking for feedback regarding a static property. A static property is not the way we generally want to add these hooks. However, we do want a more centralized access point to change these defaults, as this class is used under the hood in many different places, and it seemed this behavior did not make sense as a "client option."

Do we have best practices for how to expose such a hook for a customer to change default behavior, instead of being forced to call new overloads for every `BlobContainerClient.GetBlobClient()` call in their entire code base? Is client options still the best answer?

Reply:
Moved back to client options after testing concerns. Still happy to discuss whether this is really a "client option" but it is in clientoptions for now.